lib/logging.py

```diff
@@ -21,20 +28,30 @@ def logger() -> logging.Logger:
         return SecurityLog.seclogger
 
     @staticmethod
-    def logevent(request: HttpRequest, type: str, message: str) -> None:
-        # Unit tests do not have user defined in request object.
-        if request and hasattr(request, 'user'):
-            user=request.user
-            ip=request.META.get('REMOTE_ADDR')
-        else:
-            user='?'
-            ip='?'
-        SecurityLog.logger().info("({}/{}): {}: {}".format(
-            user,
-            ip,
-            type,
-            message
-        ))
+    def logevent(
+        request: HttpRequest,
+        type: str,
+        message: str,
+        severity: int=5,
+    ) -> None:
+        extensions = {}
+        # Unit tests may not have user or source address defined.
+        if request:
+            if hasattr(request, 'user'): extensions['sourceUserName'] = str(request.user)
```

Comment:
We could include also `sourceUserId`. No harm in having both username and user id?